File: app/src/main/java/de/bahnhoefe/deutschlands/bahnhofsfotos/util/BitmapAvailableHandler.java
package de.bahnhoefe.deutschlands.bahnhofsfotos.util;

import android.graphics.Bitmap;

import androidx.annotation.Nullable;

/**
 * Callback for BitmapDownloader
 */
public interface BitmapAvailableHandler {

    /**
     * This gets called if the requested bitmap is available. Finish and issue the notification.
     *
     * @param bitmap the fetched Bitmap for the notification. May be null
     */
    void onBitmapAvailable(@Nullable Bitmap bitmap);
}

File: app/src/main/kotlin/de/bahnhoefe/deutschlands/bahnhofsfotos/util/BitmapAvailableHandler.kt
package de.bahnhoefe.deutschlands.bahnhofsfotos.util

import android.graphics.Bitmap

/**
 * Callback for BitmapDownloader
 */
fun interface BitmapAvailableHandler {
    /**
     * This gets called if the requested bitmap is available. Finish and issue the notification.
     *
     * @param fetchedBitmap the fetched Bitmap for the notification. May be null
     */
    fun onBitmapAvailable(fetchedBitmap: Bitmap?)
}